src/test/java/seedu/address/model/tier/TierTest.java

```diff
@@ -17,10 +19,56 @@ public void constructor_invalidTierName_throwsIllegalArgumentException() {
         assertThrows(IllegalArgumentException.class, () -> new Tier(invalidTagName));
     }
 
+    @Test
+    public void constructor_emptyTierName_naTierReturned() {
+        Tier tier = new Tier("");
+        assertTrue(tier.equals(new Tier("NA")));
+    }
+
     @Test
     public void isValidTierName() {
```

Comment:
Should we test the case where the tag name is 'NA'? I was checking your test, and `Tier.isValidTierName("NA") `returns true, as expected (since it's in the enums). But should we signal that 'NA' isn't meant for external use? We implied Tier.isValidName() checks external strings, and 'NA' is just a placeholder for someone without a tier.

Reply:
This is no rush for now as it is a very architecture-related issue. We can discuss further for it sometimes.